index.js

```diff
@@ -297,21 +371,9 @@ MediaSession.prototype = extend(MediaSession.prototype, {
         this.send('source-remove', desc);
         this.pc.removeStream(stream);
 
-        this.pc.handleOffer({
-            type: 'offer',
-            jingle: this.pc.remoteDescription
-        }, function (err) {
-            if (err) {
-                self._log('error', 'Could not process offer for removing stream');
-                return cb(err);
-            }
-            self.pc.answer(self.constraints, function (err) {
-                if (err) {
-                    self._log('error', 'Could not process answer for removing stream');
-                    return cb(err);
-                }
-                cb();
-            });
+        var errorMsg = 'Could not process offer for removing stream';
+        offerAnswerQueue(self, errorMsg, this.pc.remoteDescription, function(err) {
+          cb(err);
```

Comment:
nit: to make the logic more clear (that its not just an error callback) it might be worth changing this to the code below for this and the other couple usages.
```
if(err) { 
  cb(err); 
} else { 
  cb(); 
}
```